omnigibson/examples/robots/import_custom_robot.py

```diff
@@ -0,0 +1,981 @@
+            "joint_type": "FIXED",
+            "joint_name": f"{attached_obj_link_name}_joint",
+        }
+        default_generated_cfg["collision_link_names"].append(attached_obj_link_name)
+        default_generated_cfg["extra_collision_spheres"][attached_obj_link_name] = 32
+        for link_name in gripper_info["links"]:
+            if link_name not in default_generated_cfg["self_collision_ignore"]:
+                default_generated_cfg["self_collision_ignore"][link_name] = []
+            default_generated_cfg["self_collision_ignore"][link_name].append(attached_obj_link_name)
+        default_generated_cfg["mesh_link_names"].append(attached_obj_link_name)
+
+    # Save generated file
+    Path(save_dir).mkdir(parents=True, exist_ok=True)
+    save_fpath = f"{save_dir}/{robot_name}_description_curobo_default.yaml"
+    with open(save_fpath, "w+") as f:
+        yaml.dump({"robot_cfg": {"kinematics": default_generated_cfg}}, f)
+
+    # Permute the default config to have additional base only / arm only configs
+    # Only relevant if robot is holonomic
+    if is_holonomic:
+        # Create base only config
+        base_only_cfg = deepcopy(default_generated_cfg)
+        base_only_cfg["ee_link"] = root_link
+        base_only_cfg["link_names"] = []
+        for jnt_name, default_q in joint_to_default_q.items():
+            if jnt_name not in base_only_cfg["lock_joints"] and "base_footprint" not in jnt_name:
+                # Lock this joint
+                base_only_cfg["lock_joints"][jnt_name] = default_q
+        save_base_fpath = f"{save_dir}/{robot_name}_description_curobo_base.yaml"
+        with open(save_base_fpath, "w+") as f:
+            yaml.dump({"robot_cfg": {"kinematics": base_only_cfg}}, f)
+
+        # Create arm only config
+        arm_only_cfg = deepcopy(default_generated_cfg)
+        for jnt_name in {"base_footprint_x_joint", "base_footprint_y_joint", "base_footprint_rz_joint"}:
+            arm_only_cfg["lock_joints"][jnt_name] = 0.0
+        save_arm_fpath = f"{save_dir}/{robot_name}_description_curobo_arm.yaml"
+        with open(save_arm_fpath, "w+") as f:
+            yaml.dump({"robot_cfg": {"kinematics": arm_only_cfg}}, f)
+
+
+@click.command(help=_DOCSTRING)
+@click.option(
+    "--config",
+    required=True,
+    type=click.Path(exists=True, dir_okay=False),
+    help="Absolute path to robot URDF file to import",
+)
+def import_custom_robot(config):
+    # Load config
+    with open(config, "r") as f:
+        cfg = Dict(yaml.load(f, yaml.Loader))
+
+    # Convert URDF -> USD
+    urdf_path, usd_path, prim = import_og_asset_from_urdf(
+        category="robot",
+        model=cfg.name,
+        urdf_path=cfg.urdf_path,
+        collision_method=cfg.collision.decompose_method,
+        coacd_links=cfg.collision.coacd_links,
+        convex_links=cfg.collision.convex_links,
+        no_decompose_links=cfg.collision.no_decompose_links,
```

Comment:
what is `no_decompose_links`?

Reply:
all the values are annotated in the docstring + config. These are links that should not be decomposed by either coacd / convex, and should be taken as is. It's usually useful when an explicit primitive is defined (e.g.: sphere / cylinder / box) in the URDF